### PR TITLE
new label_placer_middle function

### PR DESCRIPTION
**Pull request by jamarav**
This is a new simple label placement strategy that prints the isoline labels at the middle part of each isoline. It could be useful if the `label_placer_minmax` strategy gets bad results.

### CODE REVIEW

**Comment by clauswilke** [comment]
Thanks for the PR. A couple of quick comments (I may have more after a more careful review):

- Please follow the tidyvere style guide for code formatting. You can use the styler package (https://github.com/r-lib/styler) to automate this.
- Please use `<-` for assignments consistently.
- Please add some unit tests here: https://github.com/wilkelab/isoband/blob/master/tests/testthat/test-label-placer.R
- Not sure you need the extra assignment to `xave` and `yave`. Why not just write `xtheta - mean(xtheta)`?
- Something seems wrong with the comment in line 218. Please proof-read.

File: R/label-placer.R [hunk 1]
```diff
@@ -201,6 +203,46 @@ label_placer_manual <- function(breaks, x, y, theta) {
 }
 
 
+#' @param rot_adjuster Function that standardizes the rotation angles of the labels.
```
Comment (clauswilke) [inline R/label-placer.R:206]: I believe this argument is already documented and doesn't need to be documented again.

**Comment by jamarav** [comment]
Thanks for your comments @clauswilke. I just pushed a new commit fixing the format errors and adding  some unit tests. I also removed the extra assigments xave and yave. This extra assigments are also defined in the `minmax_impl` function. I have not modified this function but it is possible to apply the same changes.

File: R/label-placer.R [hunk 2]
```diff
@@ -201,6 +203,46 @@ label_placer_manual <- function(breaks, x, y, theta) {
...
+
+    # Then the label is printed at the middle of each isoline subdivision.
+    for (i in 1:length(line_sections)) {
+      x <- line_data$x[line_data$id == i]
+      y <- line_data$y[line_data$id == i]
+
+      middle_index <- as.integer(length(x) / 2)
+
+      x_ <- x[middle_index]
+      y_ <- y[middle_index]
```
Comment (clauswilke) [inline R/label-placer.R:225]: I don't particularly like `x_` and `y_`. How about `x_mid` and `y_mid` or similar?

**Review by clauswilke — COMMENTED** [review]
Two more minor comments, otherwise looks good. Thanks!

**Comment by clauswilke** [comment]
In `minmax_impl` the same means are used twice, so it makes sense to precalculate and store in a variable. In your function, I believe the means are used only once, so we don't need the additional assignment.

File: R/label-placer.R (hunk 1 above)
Comment (jamarav) [reply R/label-placer.R:206]: Yes it is. I'm going to removed

File: R/label-placer.R (hunk 2 above)
Comment (jamarav) [reply R/label-placer.R:225]: Ok. I will rename these two variables.

**Comment by clauswilke** [comment]
Apologies, I should have started the CI runs earlier (I always forget that they need to be started manually now). The tests are failing because the documentation needs to be rebuilt. Could you run `devtools::document()` and commit the updated documentation?

File: R/label-placer.R [hunk 3]
```diff
@@ -201,6 +203,44 @@ label_placer_manual <- function(breaks, x, y, theta) {
 }
 
 
+#' @rdname label_placer
+#' @export
+label_placer_middle <- function(rot_adjuster = angle_halfcircle_bottom(), ...) {
```
Comment (clauswilke) [inline R/label-placer.R:208]: CI still doesn't pass. It looks to me like the `...` argument here shouldn't be there. (The tests complain about an undocumented argument, but I think the bigger issue is that it doesn't belong here.)

You can run R check locally before making the next commit to make sure all issues are addressed.

Comment (jamarav) [reply R/label-placer.R:208]: yes, this is the problem.

**Comment by clauswilke** [comment]
Thanks!

**Pull request merged**